app/src/main/java/org/fcitx/fcitx5/android/input/keyboard/BaseKeyboard.kt

```diff
@@ -186,10 +198,85 @@ abstract class BaseKeyboard(
                 swipeEnabled = true
                 swipeRepeatEnabled = true
                 swipeThresholdX = selectionSwipeThreshold
-                swipeThresholdY = disabledSwipeThreshold
+                swipeThresholdY =
+                    if (backspaceSwipeClear.getValue()) inputSwipeThreshold else disabledSwipeThreshold
+                backspaceKeys += this
+                var clearTriggered = false
+                var previewShown = false
+                var pressDown = false
+                var showRunnable: Runnable? = null
```

Comment:
It's too messy to maintain those state variables here inside BaseKeyboard#createKeyView. From what it does below, I think it could be a longpress handler.